### ISSUE TITLE
FamilyDocument.GetAssociated output not correct

### ISSUE THREAD

**Issue by maxhuetten** [erfajo/OrchidForDynamo#235]
### Operating system
Windows 10

### Revit and Dynamo version
(_Which version of Revit, Dynamo and Orchid are you using? Go to Help > About if you're not sure. for Orchid version, go to Packages > Manage Packages_)
Revit: 24.1.11.26
Dynamo: v.2.18.1
Orchid: 2.18.1.6034

### What did you do? 
I am using the FamilyDocument.GetAssociated node from orchid to retrieve all elements which can get parameters associated inside of familyDocuments, which are opened in the backround (Document.BackroundOpen from orchid).

### What did you expect to see?
I was expecting an output of all elements which allow parameters to be associated plus a list of list of these parameters (once their names and once in their parameter form).

### What did you see instead?
The output of elements is working just fine, i am able to retrieve all connector elements inside my families but i only get one parameter for each of these elements. And these are not, as the tooltip suggests, the associated parameters in the document. But only a selection of them.
![85986c10c4c6255b8a3279fa072a3bf948fd63ab](https://github.com/erfajo/OrchidForDynamo/assets/160142655/eda3fb57-0625-4e72-a54c-bbad6ab4ce4e)

**Comment by erfajo** [comment]
The family document can only have one associated element to a parameter...
If you want to use the parameters inside the associated elements, then use those elements. the parameter values are shown, but you can't use them.

**Comment by maxhuetten** [comment]
Alright, i expected a different output of the node.

If i understand you correct the node is looking from a different POV. It is not retrieving elements which have associated parameters but it is retrieving every parameter which is associated. Do i get this right?

Secondly, if this is the way it should be working, it still isn't for me. For example i have a Venting Family which has 2 connector Elements, each has 3 associated parameters (height, width und flowrate), all 3 of them are associated to the same parameters. When i use the node on the family document i get only 5 parameters and not even the correct ones.
![image](https://github.com/erfajo/OrchidForDynamo/assets/160142655/99cc8114-ad8c-4ab7-8eb7-ce78fae28ae6)

I might missunderstand the word asscoiated in the family context..

**Comment by erfajo** [comment]
Let me try to clarify. I will use the sample graph ”Nested-Lookup” (comes in both the new layout Revit 23 and the older layout below Revit 22), and how it looks inside Revit.

The associated nodes look up families inside the family document, in this case, a casework plain front, this nested family has a parameter ”width” there has been associated with the family document parameter ”width”. Meaning, if the family document parameter ”width” is changed from 800 to 100, then will the parameter value ”width” in the nested family ”Front – Plain” be changed.

You can either look up using the parameter in the family document, find exactly which associated families are connected, and get the parameter, or you can get all associations in the family document, in that case, you will get the family document parameter, the associated family(ies), and the parameter in the family.

In other words, a parameter in a nested family can only have one family document parameter value associated. However, more than one nested family can be associated with a family document parameter. But this parameter can hold one value. In this case, is the value 800. This value will be sent to all those nested family parameters that subscribe to the family document parameter.

Does this give you an idea of how it works? And please try to download the sample graph so you can test it by your selves.
![image](https://github.com/erfajo/OrchidForDynamo/assets/2844374/de1c24ae-cf80-4e15-a76a-1c77543ce95f)

**Comment by maxhuetten** [comment]
Thank you for the thorough explanation.
I think i understood how FamilyDocument.GetAssociated works. And that i was looking for FamilyParameter.GetAssociated, to find all families which are connected to the parameter. 
Still if two, or more, families are connected to the same parameter, i.e. airflow, only one pops up using the node. Is this intended?

**Comment by erfajo** [comment]
now I get it... you are right, you should be able to get all the associated parameters!?
I will try to see how I can manage that. However, I am refactoring my Material nodes at the moment, and I have to test them before I can release anything new... hopefully within 5-7 days.

**Comment by maxhuetten** [comment]
Thank you very much! 
Sorry for the confusion and my lack of understanding how your nodes work.

I am looking forward to testing out the new update in a few weeks!

**Comment by erfajo** [comment]
This one must be on me, I was me who didn't understand :-)
You have contributed so the node is much better now :-)

**Comment by erfajo** [comment]
I have uploaded a new build (6167) today, this includes the solution for this issue :-)
